### DISCUSSION TITLE
adaptive updates - failed to open target slot hash index

### DISCUSSION THREAD

**Discussion by ghost** [rauc/rauc#1191] [General]
Also using the journalctl -u rauc why do we get this error and how do we resolve it

> Jul 21 08:49:43 imx8mnevk rauc[111101]: [[0;1;38;5;185m[[0;1;39m[[0;1;38;5;185mContinuing after adaptive mode error: failed to open target slot hash index for rootfs.1: data file size (1621870592) is not a multiple of 4096 bytes[[0m

**Comment by ejoerns** [comment]
@pritpalm-kuk The error message is probably not too self-explaining at this level. What it should mean is that the size of the target slot is not a multiple of 4096 and can thus not be used for adaptive updates.

Could you verify that this is actually the case? For fixing, you would need to make the partition size a multiple of 4096.

**Comment by ghost** [comment]
Hi

Thanks for the update
The partition alignment is set to 8192. It has to be that suze cause our
architecture is 64but

We have no max suze on the partition

Regards

Pritpal

On Mon, 31 Jul 2023 at 15:46, Enrico Jorns ***@***.***> wrote:

> @pritpalm-kuk <https://github.com/pritpalm-kuk> The error message is
> probably not too self-explaining at this level. What it should mean is that
> the size of the target slot is not a multiple of 4096 and can thus not be
> used for adaptive updates.
>
> Could you verify that this is actually the case? For fixing, you would
> need to make the partition size a multiple of 4096.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/rauc/rauc/discussions/1191#discussioncomment-6594803>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ANVR26AOTCZZWHWXPXIASKLXS6SJXANCNFSM6AAAAAA2TA4USI>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

-- 
This e-mail is intended only for the person to whom it is addressed and may 
contain confidential information. If an addressing or transmission error 
has misdirected this e-mail, please notify the author by replying to this 
e-mail. If you are not the intended recipient you must not use, disclose, 
print or rely on the contents of this e-mail and should immediately and 
permanently delete it.


Kenwood Limited is a company registered in England 
with registered number 872044. The registered office of Kenwood Limited is 
1 Kenwood Business Park, New Lane, Havant, Hampshire, PO9 2NH.


Kenwood 
Limited may monitor outgoing and incoming e-mails and other 
telecommunications on its e-mail and telecommunications systems for the 
purpose of ensuring compliance with the law and with Kenwood Limited's 
policies.

**Reply by ejoerns** [reply]
> Thanks for the update The partition alignment is set to 8192. It has to be that suze cause our architecture is 64but

I was not referring to the partition alignment but the partition size. A partition must have a size. Could you check which size it is?

> We have no max suze on the partition

Could you clarify what is meant by this?

**Comment by ghost** [comment]
Hi

Thanks for the update. Please have a look at the fragment below from our
.wic file

```
part u-boot --source rawcopy --sourceparams="file=imx-boot" --ondisk mmcblk --no-table --align 32
part /boot --source bootimg-partition --ondisk mmcblk --fstype=vfat --label boot --active --align 8192 --size 64
part / --source rootfs --ondisk mmcblk --fstype=ext4 --label rootfs_A --align 8192
part / --source rootfs --ondisk mmcblk --fstype=ext4 --label rootfs_B --align 8192
part /data --source rootfs --rootfs-dir=${IMAGE_ROOTFS}/data --ondisk mmcblk --fstype=ext4 --size 100K --label data --align 8192
part /fresco --source rootfs --rootfs-dir=${IMAGE_ROOTFS}/fresco --ondisk mmcblk --fstype=ext4 --size 10K --label fresco --align 8192
```

As you can see we have not set a —size (min size) or —fixed-size (exact partition size)

So are you wanting us to set a —fixed-size


Regards

Pritpal


On Tue, 1 Aug 2023 at 09:14, Enrico Jorns ***@***.***> wrote:

> Thanks for the update The partition alignment is set to 8192. It has to be
> that suze cause our architecture is 64but
>
> I was not referring to the partition alignment but the partition size. A
> partition must have a size. Could you check which size it is?
>
> We have no max suze on the partition
>
> Could you clarify what is meant by this?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/rauc/rauc/discussions/1191#discussioncomment-6601872>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ANVR26CQLZDIZN5YMUHQC4DXTCNFJANCNFSM6AAAAAA2TA4USI>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

-- 
This e-mail is intended only for the person to whom it is addressed and may 
contain confidential information. If an addressing or transmission error 
has misdirected this e-mail, please notify the author by replying to this 
e-mail. If you are not the intended recipient you must not use, disclose, 
print or rely on the contents of this e-mail and should immediately and 
permanently delete it.


Kenwood Limited is a company registered in England 
with registered number 872044. The registered office of Kenwood Limited is 
1 Kenwood Business Park, New Lane, Havant, Hampshire, PO9 2NH.


Kenwood 
Limited may monitor outgoing and incoming e-mails and other 
telecommunications on its e-mail and telecommunications systems for the 
purpose of ensuring compliance with the law and with Kenwood Limited's 
policies.

**Reply by ejoerns** [reply]
Yes, please use `--fixed-size` if you use the wic image for bootstrapping your device's storage.

Otherwise, the size of your partition will always depend on the size of the rootfs and the configured overhead.
This will produce unexpected results if you bootstrap different devices with different versions of the wic image.
Instead, I would propose to figure out a proper size for your rootfs partitions (with enough space left for future rootfs/application extensions and updates).

**Comment by ghost** [comment]
We have solved the issue

**Reply by ejoerns** [reply]
Did `--fixed-size` make it? Or was more required? Would be great to share how you fixed it (for others).

**Reply by ghost** [reply]
Yes using the --fixed-size solved our issue

**Reply by ghost** [reply]
We are wanting to use the busctl get-property de.pengutronix.rauc command to filter on Operation Progress LastError as shown below
busctl get-property de.pengutronix.rauc / de.pengutronix.rauc.Installer Operation
busctl get-property de.pengutronix.rauc / de.pengutronix.rauc.Installer Progress
busctl get-property de.pengutronix.rauc / de.pengutronix.rauc.Installer LastError

Please can we have a list all return types for the following
Operation
Progress
LastError

**Reply by ejoerns** [reply]
Please do not mix in too many questions here. Feel free to open another discussion.

**Reply by ejoerns** [reply]
Ah, saw you already did. So no double-post please.

**Reply by ghost** [reply]
ok sorry
